splink/settings.py

```diff
@@ -155,6 +156,38 @@ def _needs_matchkey_column(self) -> bool:
 
         return len(self._blocking_rules_to_generate_predictions) > 1
 
+    def _generate_blank_salts(self, rules=[]):
+        return _add_salt_to_blocking_rules(rules)
+
+    def _generate_blocking_rules_to_generate_predictions(
+        self, blank=False
+    ) -> List[str]:
+
+        if blank:
+            self._blocking_rules_to_generate_predictions = []
+            return
+
+        # salts are set to the input rules within the settings dictionary
+        # if no salting is selected
+        rules = self._from_settings_dict_else_default(
+            "blocking_rules_to_generate_predictions"
+        )
+
+        self._splink_salts = _add_salt_to_blocking_rules(rules, self._salting)
+        self._blocking_rules_to_generate_predictions = self._splink_salts.keys()
```

Comment:
the `_splink_salts` key and the `_blocking_rules_to_generate_predictions` are created to work together, but they exist separately.

I think we need some mechanism to 'bind' these two things together.

I'll do a longer comment on the main PR

Reply:
This should now all be incorporated into the new [BlockingRule class](https://github.com/moj-analytical-services/splink/pull/600/commits/b3c373f746c5e7cfac10a5b23015dd34e7b4a87f).